tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,240 @@
+				if ocp == "true" {
+					extraArg = "--set=platform=openshift"
+				}
+
+				installArgs := fmt.Sprintf("--namespace %s --set=image=%s %s", namespace, image, extraArg)
+				Eventually(helm.Install).
+					WithArguments("sail-operator", filepath.Join(baseDir, "chart"), installArgs).
+					Should(Succeed(), "Operator failed to be deployed; unexpected error")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetJSONCondition).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable))
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=istio-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("Istio-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(istioResources), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+			var err error
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+					istioCR = v1alpha1.Istio{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: "operator.istio.io/v1alpha1",
+							Kind:       "Istio",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: istioName,
+						},
+						Spec: v1alpha1.IstioSpec{
+							Namespace: controlPlaneNamespace,
+							Version:   version,
+						},
+					}
+
+					istioCRYAML, err = yaml.Marshal(istioCR)
+					Expect(err).ToNot(HaveOccurred(), "Istio CR failed to be created; unexpected error")
+				})
+
+				When("the resource is created", func() {
+					Specify("successfully", func() {
+						Eventually(kubectl.ApplyString).
+							WithArguments(string(istioCRYAML)).
+							Should(Succeed(), "Istio CR failed to be created; unexpected error")
+						Success("Istio CR created")
+					})
+
+					It("updates the Istio resource status to Reconcilied and Ready", func() {
+						Eventually(kubectl.GetJSONCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconcilied), "Istio it's not Reconcilied; unexpected Condition")
+
+						Eventually(kubectl.GetJSONCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio it's not Ready; unexpected Condition")
+
+						Eventually(kubectl.GetPodPhase).
+							WithArguments(controlPlaneNamespace, "app=istiod").
+							Should(Equal("Running"), "Istiod is not running; unexpected pod Phase")
```

Comment:
This is in the wrong place. It belongs under `It("deploys istiod"`, but I think we could simply remove it, since the test only needs to check that the istiod deployment is created. Everything else must be done by Kubernetes.